StRoot/StFstRawHitMaker/StFstRawHitMaker.cxx

```diff
@@ -425,19 +425,7 @@ Int_t StFstRawHitMaker::Make()
 
                     for( rawHitIter = rawHitVec.begin(); rawHitIter != rawHitVec.end(); ++rawHitIter ){
                         StFstRawHit* rawHit = *rawHitIter;
-
-                        StFstRawHit* newRawHit = new StFstRawHit();
-                        newRawHit->setChannelId( rawHit->getChannelId() );
-                        newRawHit->setGeoId( rawHit->getGeoId() );
-                        newRawHit->setSeedhitflag( rawHit->getSeedhitflag() );
-                        for(int itb=0; itb<kFstNumTimeBins; itb++){
-                            newRawHit->setCharge( rawHit->getCharge(itb), itb );
-                            newRawHit->setChargeErr( rawHit->getChargeErr(itb), itb );
-                        }
-                        newRawHit->setMaxTimeBin( (int)rawHit->getMaxTimeBin() );
-                        newRawHit->setDefaultTimeBin( (int)rawHit->getDefaultTimeBin() );
-                        newRawHit->setIdTruth( rawHit->getIdTruth() );
-
+                        StFstRawHit* newRawHit = new StFstRawHit( rawHit );
```

Comment:
Are you actually sure you need to create copies of StFstRawHit objects with the `new` operator? Can't you just copy pointers from `rawHitCollectionPtr->getRawHitVec()` to `mFstEvtCollection->rawHits()`?

Reply:
I had tried to directly put `rawHit`, the pointer from the iterator of `mFstEvtCollection->rawHits()` before as following:
```
mFstEvtCollection->addRawHit(rawHit);
```
The compiler won't complain but I got some error like:
```
StEventMaker:FATAL - TUnixSystem::DispatchSignals : segmentation violation
```
when I ran the codes.